### PR TITLE
Add Native Asset Support and Enhance Unit Testing

### LINKED ISSUE #2: Minor: `decode` does not validate HRP conformance to Bech32 character constraints
Minor: The `decode` does not validate that the HRP conforms to Bech32 character constraints (a-z, 0-9). This could lead to issues if an HRP with invalid characters is used.

Potential validation could be done using the following regex pattern `/^[a-z0-9]+$/`

_Originally posted by @latheesan-k in https://github.com/cardano-php/bech32/pull/1#discussion_r1901185127_

### LINKED ISSUE #8: Add support for Native Asset encoding/decoding
[CIP-0014](https://github.com/cardano-foundation/CIPs/tree/6424b4fd8c76e61b78feb303182a7aa61453e78d/CIP-0014) defines a method for Bech32-encoding and decoding Cardano Native Assets.

[Test Vectors](https://github.com/cardano-foundation/CIPs/tree/6424b4fd8c76e61b78feb303182a7aa61453e78d/CIP-0014#test-vectors) are also defined so that we can use Unit Testing to prove that our implementation matches the existing JS and Haskell implementations.

### PR DESCRIPTION
**Pull request by Crypto2099**
Add support and helper functions and unit tests for encoding and decoding Cardano Native Assets into an "Asset Fingerprint" which is a more human-friendly identifier defined in [CIP-0014](https://github.com/cardano-foundation/CIPs/tree/6424b4fd8c76e61b78feb303182a7aa61453e78d/CIP-0014).

- Resolves #8 
- Fixes #2

### CODE REVIEW

File: tests/Bech32AssetTest.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace CardanoPhp\Bech32;
+
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\Attributes\DataProvider;
+use PHPUnit\Framework\TestCase;
+
+#[CoversClass(Bech32::class)]
+class Bech32AssetTest extends TestCase
```
Comment (latheesan-k) [inline tests/Bech32AssetTest.php:10]: Minor recommendations:

* Expand test coverage with unhappy path also
  * Add test that provide invalid input and edge cases (min/max input, zero lonth and other boundary conditions)
  * Use php unit's `assertException` helper to check expected exceptions are thrown for invalid inputs / edge cases

Comment (latheesan-k) [reply tests/Bech32AssetTest.php:10]: Ignore above recommendations, appears to be covered in the next test.

**Review by latheesan-k — APPROVED** [review]
LGTM
